linux/src/file_util_linux.c

```diff
@@ -0,0 +1,133 @@
+            {
+                //Codes_SRS_FILE_UTIL_LINUX_09_006: [ If desired_access is GENERIC_ALL or GENERIC_READ&GENERIC_WRITE, file_util_create_file will call open with O_RDWR and shall return a file handle for read and write. ]
+                user_access = O_RDWR;
+            }
+            
+            if (creation_disposition == CREATE_ALWAYS || creation_disposition == OPEN_ALWAYS)
+            {
+                //Codes_SRS_FILE_UTIL_LINUX_09_014: [ If creation_disposition is CREATE_ALWAYS or OPEN_ALWAYS, file_util_create_file will call open with O_CREAT and shall either create a new file handle if the specificied pathname exists and return it or return an existing file handle. ]
+                result_creation_disposition = O_CREAT;
+            }
+            else if (creation_disposition == CREATE_NEW)
+            {
+                //Codes_SRS_FILE_UTIL_LINUX_09_016: [ If creation_disposition is CREATE_NEW and the file already exists, file_util_create_file will fail and return INVALID_HANDLE_VALUE. ]
+                //Codes_SRS_FILE_UTIL_LINUX_09_015: [ If creation_disposition is CREATE_NEW, file_util_create_file will call open with O_CREAT|O_EXCL and shall return a new file handle if the file doesn't already exist. ]
+                result_creation_disposition = O_CREAT|O_EXCL;
+            }
+            else if (creation_disposition == TRUNCATE_EXISTING)
+            {
+                //Codes_SRS_FILE_UTIL_LINUX_09_017: [ If creation_disposition is TRUNCATE_EXISTING, file_util_create_file will call open with O_TRUNC and shall return a file handle who's size has been truncated to zero bytes. ]
+                result_creation_disposition = O_TRUNC;
+            }
+
+            int flags = user_access|result_creation_disposition;
+            result->h_file = open(full_file_name, flags);
+
+            if(result -> h_file == -1)
+            {
+                //Codes_SRS_FILE_UTIL_LINUX_09_008: [ If there are any failures, file_util_create_file shall fail and return INVALID_HANDLE_VALUE. ]
+                LogError("Failure in creating a file, full_file_name=%s", full_file_name);
+                free(result);
+                result = INVALID_HANDLE_VALUE;
+            }
+        }
+    }
+    return result;
+}
+
+//Codes_SRS_FILE_UTIL_LINUX_09_011: [ file_util_close_handle closes a file handle in linux and returns true. ]
+bool file_util_close_handle(HANDLE handle_input)
+{
+    bool result;
+
+    //Codes_SRS_FILE_UTIL_LINUX_09_009: [ If handle_input is NULL, file_util_close_handle returns false. ]
+    if(handle_input == INVALID_HANDLE_VALUE)
+    {   
+        result = false;
+        LogError("Invalid argument to file_destroy. Handle input was NULL");
+    }
+    else 
+    {
+        CREATE_FILE_LINUX* cfl = (CREATE_FILE_LINUX*)handle_input;
+
+        if(close(cfl->h_file) == 0)
+        {
+            //Codes_SRS_FILE_UTIL_LINUX_09_013: [ If close returns a zero, then file_util_close_handle shall return true. ]
```

Comment:
>//Codes_SRS_FILE_UTIL_LINUX_09_013: [ If close returns a zero, then file_util_close_handle shall return true. ] [](http://example.com/codeflow?start=12&length=111)

Requirements 13 and 12 dont need to be conditional on the return value of close. You just need one requirement that says:

"""
`file_util_close_handle` shall call `close` on the given `handle_input`.
"""

The success and failure conditions will be covered by requirements such as:

`file_util_close_handle` shall succeed and return `true`.

`file_util_close_handle` shall fail and return `false`.